### PR TITLE
Merge r1.16.0 release main

### PR DESCRIPTION
**Pull request by ericharper**
# What does this PR do ?

Merges bugfixes and doc updates from the release branch into main.

(these should be done automatically by the github action but this PR will make sure the release branch can be rebased onto main)

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: tutorials/tts/NeMo_TTS_Primer.ipynb [hunk 1]
```diff
@@ -129,7 +129,7 @@
     "While this is the most common structure, there may be fewer or additional steps depending on the use case. For example, some languages do not require G2P and can instead rely on the model to convert raw text/graphemes to spectrogram.\n",
     "\n",
     "<div>\n",
-    "<img src=\"https://raw.githubusercontent.com/NVIDIA/NeMo/main/tutorials/tts/images/tts_pipeline.png\" width=\"500\", height=\"500\"/>\n",
+    "<img src=\"https://raw.githubusercontent.com/NVIDIA/NeMo/stable/tutorials/tts/images/tts_pipeline.png\" width=\"500\", height=\"500\"/>\n",
```
Comment (XuesongYang) [inline tutorials/tts/NeMo_TTS_Primer.ipynb:132]: One question about which branch should be applied cause I am not sure about our convention:
1. choose `main` or `stable` for `main` branch tutorials?
2. choose `release version` for `release` branch tutorials?

Comment (XuesongYang) [reply tutorials/tts/NeMo_TTS_Primer.ipynb:132]: discussed offline. links should always be `stable` no matter it is `main` or `release` branch.

**Review by XuesongYang — APPROVED** [review]
LGTM for TTS tutorials. Let's wait other folk approval. Thanks!

File: tutorials/tts/NeMo_TTS_Primer.ipynb (hunk 1 above)
Comment (titu1994) [reply tutorials/tts/NeMo_TTS_Primer.ipynb:132]: Hmm thats not the case. Main branch notebooks should have main as their branch, release branch should have stable as their branch when selecting version to install inside the notebook. When release branch is cut, this is changed from "main" to "rABC.XYZ" 

See https://github.com/NVIDIA/NeMo/blob/r1.16.0/tutorials/00_NeMo_Primer.ipynb and https://github.com/NVIDIA/NeMo/blob/main/tutorials/00_NeMo_Primer.ipynb

**Review by titu1994 — APPROVED** [review]
Thanks ! Looks great !

File: tutorials/tts/NeMo_TTS_Primer.ipynb (hunk 1 above)
Comment (XuesongYang) [reply tutorials/tts/NeMo_TTS_Primer.ipynb:132]: hmm...I was advised that links should always be `stable`. @ericharper may provide a better explanation here.

**Pull request merged**